core/api/src/main/java/org/trellisldp/api/ImmutableDataService.java

```diff
@@ -34,8 +34,8 @@
      * @param identifier the identifier under which to persist a dataset
      * @param session the session context for this operation
      * @param dataset a dataset to persist
-     * @return whether the resource was successfully persisted
+     * @return the new completion stage
```

Comment:
I think we need a stronger comment than this. We need to explain what returning various values means. E.g. a completed return indicates that persistence  completed, with errors appearing as exceptions in the completed future.

Reply:
Good point. I'll improve the javadocs